modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/daemon/DownloadDaemon.scala

```diff
@@ -3,25 +3,34 @@ package org.tessellation.sdk.infrastructure.snapshot.daemon
 import cats.effect.Async
 import cats.effect.std.Supervisor
 import cats.syntax.eq._
+import cats.syntax.flatMap._
 import cats.syntax.functor._
 
 import org.tessellation.schema.node.NodeState
 import org.tessellation.sdk.domain.Daemon
 import org.tessellation.sdk.domain.node.NodeStorage
+import org.tessellation.sdk.domain.snapshot.PeerDiscoveryDelay
 import org.tessellation.sdk.domain.snapshot.programs.Download
 
 trait DownloadDaemon[F[_]] extends Daemon[F] {}
 
 object DownloadDaemon {
 
-  def make[F[_]: Async](nodeStorage: NodeStorage[F], download: Download[F])(implicit S: Supervisor[F]) = new DownloadDaemon[F] {
+  def make[F[_]: Async](
+    nodeStorage: NodeStorage[F],
+    download: Download[F],
+    downloadDelay: PeerDiscoveryDelay[F]
```

Comment:
it's a peer discovery delay (executed before download). I'd not name it `download` to keep it uncoupled